app/policies/trainee_policy.rb

```diff
@@ -22,27 +22,31 @@ def initialize(user, trainee)
   end
 
   def show?
-    user && (user.system_admin? || user.provider_id == trainee.provider_id)
-  end
-
-  def confirm?
-    user && (user.system_admin? || user.provider_id == trainee.provider_id)
-  end
-
-  def recommended?
-    user && (user.system_admin? || user.provider_id == trainee.provider_id)
+    expected_user_signed_in?
   end
 
   def withdraw?
-    defer? || trainee.deferred?
+    expected_user_signed_in? && (defer? || trainee.deferred?)
   end
 
   def defer?
-    trainee.submitted_for_trn? || trainee.trn_received?
+    expected_user_signed_in? && (trainee.submitted_for_trn? || trainee.trn_received?)
+  end
+
+  def reinstate?
+    expected_user_signed_in? && trainee.deferred?
   end
 
   alias_method :create?, :show?
   alias_method :update?, :show?
   alias_method :edit?, :show?
   alias_method :new?, :show?
+  alias_method :confirm?, :show?
+  alias_method :recommended?, :show?
+
+private
+
+  def expected_user_signed_in?
+    user && (user.system_admin? || user.provider_id == trainee.provider_id)
+  end
```

Comment:
It's kinda saying we will have an unexpected unsigned in user, probably fussy from me

maybe
```suggestion
  def allowed_user?
    user&.system_admin? || user&.provider == trainee.provider
  end
```

Reply:
Ha! No you're definitely right, I struggled with thinking of a name, `allowed_user` is much better 👍